src/consensus/pbft/libbyz/Node.cpp

```diff
@@ -375,4 +375,6 @@ void Node::set_f(ccf::NodeId f)
   LOG_INFO << "***** setting f to " << f << "*****" << std::endl;
   max_faulty = f;
   send_only_to_self = (f == 0);
-}
+  threshold = f * 2 + 1;
+  num_replicas = 3 * f + 1;
```

Comment:
It is at this point it is true in the future it may change but i am not sure exactly how. There are 2 schools of thought: 1 - f is set by the members ( current implementation) , 2 - f is derived from the number of replicas.